fili-core/src/main/java/com/yahoo/bard/webservice/data/dimension/impl/LuceneSearchProvider.java

```diff
@@ -318,6 +327,133 @@ private void updateDimensionRow(
     }
 
     /**
+     * Replaces Lucene indexes with a set of new indexes in hot-swap manner.
+     *
+     * @param newLuceneIndexPathString  The location of directory that contains the the new indexes.
+     */
+    public void replaceIndex(String newLuceneIndexPathString) {
+        lock.writeLock().lock();
+        try {
+            String source = luceneIndexPath;
+            String destination = luceneIndexPath.substring(
+                    0,
+                    luceneIndexPath.length() - (luceneIndexPath.endsWith("/") ? 1 : 0)
+            ) + "_old";
+
+            // move Lucene index directory to a different location under the same parent
+            moveDir(source, destination);
+            LOG.debug("Old Lucene index dir has been moved from {} to {}", source, destination);
+
+            // move all files under new lucene index directory to the old directory
+            LOG.debug("Moving all new Lucene indexes from {} to {}...", newLuceneIndexPathString, luceneIndexPath);
+            try {
+                Files.createDirectory(new File(luceneIndexPath).getAbsoluteFile().toPath());
+            } catch (IOException e) {
+                LOG.error(ErrorMessageFormat.CANNOT_CREATE_INDEX_DIR.format(luceneIndexPath));
+                throw new RuntimeException(e);
+            }
+            moveDirEntries(newLuceneIndexPathString, luceneIndexPath);
+
+            // file system delete old Lucene index directory
+            deleteDir(destination);
```

Comment:
Perhaps change the variable name destination to indicate that it's the tempDirectory to be deleted?